### PR TITLE
complete how to npm

### PR DESCRIPTION
**Pull request by msyao**

### CODE REVIEW

**Comment by msyao** [comment]
& completed up to 08 in **learnyounode**

**Comment by msyao** [comment]
completed express works

File: expressworks/param.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  // could have more than 1 params, ie /message/:id/:name/:data
+  var str = require('crypto')
+  .createHash('sha1')
+  .update(new Date().toDateString() + id)
+  .digest('hex')
+  res.send(str)
+
+});
+
+app.listen(process.argv[2])
```
Comment (GA-MEB) [inline expressworks/param.js:16]: Very nice.

File: learnyounode/04-filtered-ls.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+var fs = require('fs')
+var path = require('path')
+
+fs.readdir(process.argv[2], function (err, list) {
+  list.forEach(function (file) {
+    if (path.extname(file) === '.' + process.argv[3])
+    console.log(file)
+  })
+});
```
Comment (GA-MEB) [inline learnyounode/04-filtered-ls.js:22]: Don't forget to handle errors.

File: learnyounode/05-make-it-modular.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+var extStr = process.argv [3];
+
+mymodule(dir, extStr, function (err, data) {
+  if (err){
+    return err;
+  }
+  data.forEach(function (file) {
+    console.log(file)
+  });
+});
```
Comment (GA-MEB) [inline learnyounode/05-make-it-modular.js:12]: Nice, you aced this one.

File: learnyounode/06-http-client.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+var http = require('http')
+, url = process.argv[2];
+
+http.get(url, function (response) {
+  response.setEncoding('utf8');
+  response.on('data', function (data) {
+    console.log(data);
+  })
+});
```
Comment (MattB-GA-Boston) [inline learnyounode/06-http-client.js]: Don't forget the error case!

File: learnyounode/08-juggling-async.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+// , url = process.argv[2];
+//
+// http.get(url, function (response) {
+//   response.setEncoding('utf8');
+//   response.pipe(concatStream(function (data) {
+//     console.log(data.length);
+//     console.log(data);
+//
+//   }));
+// });
```
Comment (GA-MEB) [inline learnyounode/08-juggling-async.js:12]: This one's pretty tricky. See if you can give it another shot at some point.

**Pull request closed without merging**